doc: add guide for update and install using LACP (#385)
xcp-ng/xcp-ng-org: docs/guides/lacp-install-upgrade.md

- The recommended approach is to enable "LACP fallback" (or an equivalent feature) on your switch. This way you can simply use a single interface during installation or upgrade.
- If you’re performing a new installation on an LACP-only network, be aware that your host will lose network connectivity after booting. You’ll need physical access or IPMI access to configure the network properly afterward. Xen Orchestra won’t be able to assist with bond creation at this stage.
- For upgrades, your existing configuration will remain intact after the upgrade is done.
:::

Our installer does not support LACP. This guide explains how to bypass this limitation, but please read all the instructions carefully before proceeding, as there are important caveats.
When using LACP, your switch needs to be configured for it. If you cannot or prefer not to reconfigure your switch, you won’t be able to use the single-interface option provided by the installer.
For other bonding methods (active-backup, balance-lsb), the switch does not need to be aware of the bond, and you can simply use one of the interfaces that will be part of the bond.

> ## Do you actually need network during the installation process?

Reviewer: This title seems long, which can make navigation awkward:

<img width="2418" height="1829" alt="Capture d'écran 2025-09-04 124319" src="https://github.com/user-attachments/assets/73307ce1-7a18-410c-81ac-7e81845b615b" />

Readers should be able to scan the document quickly, to find the section they're interested in.

We can keep the title as-is if necessary, but it would be extra nice if we could find a shorter one.

Author: Done.

A working network configuration is only necessary in the following scenarios:

- New installation from net install ISO.
- Upgrading supporting hosts, even from a full ISO, as they need to contact the master host during the ugprade.
- Upgrading all hosts from net install ISO.

Network access is not required for:

- A new installation from the full ISO.
- Upgrading the master host from the full ISO.

## Installation or Upgrade steps

:::info
During an upgrade, the installer will not detect your existing bond configuration. In XCP-ng, network settings are managed by XAPI and applied to Open vSwitch, but the installation process operates independently of these components.
:::

> 1. Boot from your installation media. You'll reach the **Select Keymap** screen, at this stage, the network interfaces still have their original names, such as `side-XXXX-ethY`

Reviewer: ```suggestion
1. Boot from your installation media. You'll reach the **Select Keymap** screen, at this stage, the network interfaces still have their original names, such as `side-XXXX-ethY`.
```

:::warning
Do not attempt to configure the bond yet. Doing so would mislead the installer into thinking you have no network interfaces, which would block the installation or upgrade process.
:::

![keymap screen](../assets/img/installer-keymap-screen.png)

> 2. Select your keymap, and proceed to the next screen: **Welcome to XCP-ng Setup**

Reviewer: ```suggestion
2. Select your keymap and proceed to the next screen (**Welcome to XCP-ng Setup**).
```

3. You can now change to the second terminal using either `alt+F2` on `alt+right`
4. Check the interfaces are named `ethX` using `ip link`. If that's the case, you can proceed with the next steps.
5. Now we can configure a bond using standard linux tools.

```bash
modprobe bonding mode=802.3ad miimon=100 # automatically creates bond0 in 8.3 installer
ip link set eth0 down
ip link set eth1 down
ip link set eth0 master bond0
ip link set eth1 master bond0
```

6. Configure an IP address in one of two ways:

- Via DHCP:

```bash
dhclient bond0
```

- Manually assigning a static IP:

```bash
ip addr add aaa.bbb.ccc.ddd/NN dev bond0
ip route add default via aaa.bbb.ccc.1
ip link set bond0 up
```

For example:
![configure LACP](../assets/img/installer-creating-lacp-bond.png)

### Upgrading a supporting host

> 7. Follow the standard installation procedure as usual. When you reach the networking configuration step, you’ll see an option labeled **Use existing configuration**. Select this option and complete your upgrade.

Author: Done.

![net install management network](../assets/img/netinstall-management-network.png)

9. Proceed with the installation.
10. Reboot your host.

:::tip
The first boot can take a while. Give the host enough time for everything to start up before proceeding to the next steps.
:::

- If you used DHCP, `xsconsole` will likely display **No network configured**.
- If you used a static IP, it will show the configuration, but only on the `ethX`` interface you selected.

At this point, you’ll need to:

1. Create your bond
2. Set the bond as the management interface.

:::info
`xe` commands now feature improved tab completion. Use it to avoid running `*-list` commands to find UUIDs and retyping them manually.
:::

1. Create a new network named `bond0`. For example:

```bash
xe network-create name-label="bond0"
```

Create the bond using the network-uuid that command just returned:

```bash
xe bond-create mode=lacp network-uuid=<network_uuid> pif-uuids=<pif_uuid#1>,<pif-uuid#2>
```

> 2. Set it as your management interface:

Author: Done.

# Using LACP to Net Install or Upgrade XCP-ng

## Foreword

:::warning

> 

Author: That was from the linter, I removed them.

- If you’re performing a new installation on an LACP-only network, be aware that your host will lose network connectivity after booting. You’ll need physical access or IPMI access to configure the network properly afterward. Xen Orchestra won’t be able to assist with bond creation at this stage.
- For upgrades, your existing configuration will remain intact after the upgrade is done.

:::

Our installer does not support LACP. This guide explains how to bypass this limitation, but please read all the instructions carefully before proceeding, as there are important caveats.
When using LACP, your switch needs to be configured for it. If you cannot or prefer not to reconfigure your switch, you won’t be able to use the single-interface option provided by the installer.
For other bonding methods (active-backup, balance-lsb), the switch does not need to be aware of the bond, and you can simply use one of the interfaces that will be part of the bond.

## Do you actually need network?

A working network configuration is only necessary in the following scenarios:

- New installation from net install ISO

> - Upgrading supporting hosts, even from a full ISO

Reviewer 2: ```suggestion
- Upgrading supporting hosts, even from a full ISO, as they need to contact the master host during the ugprade
```

Author: Done, I also added periods to these bullet point now that we have an actual sentence in there.

- Upgrading all hosts from net install ISO

Network access is not required for:

- A new installation from the full ISO
- Upgrading the master host from the full ISO

## Installation or Upgrade steps

:::info
During an upgrade, the installer will not detect your existing bond configuration. In XCP-ng, network settings are managed by XAPI and applied to Open vSwitch, but the installation process operates independently of these components.
:::

> 1. Boot from your installation media. You'll reach the **Select Keymap** screen, at this point stage, the network interfaces still have their original names, such as `side-XXXX-ethY`

Author: messed up with Thomas suggestion integration, thanks for noticing it.

# Using LACP to Net Install or Upgrade XCP-ng

:::warning
**Foreword**:

- This guide is only meant for people having LACP only network, without "LACP
fallback" and don't want to reconfigure their switches.

> - The recommended way is to enable "LACP fallback" or equivalent on your switch,

Reviewer: ```suggestion
- The recommended approach is to enable "LACP fallback" (or an equivalent feature) on your switch.
```

so you can simply use a single interface during installation or upgrade.

> - If you follow this guide for a new net install, on a network that is LACP

Author: Done, I used your sentence as is.

only, keep in mind you won't have any network once your host is booted up, so
you will need physical access to the host or some kind of IPMI access to
configure your network properly after the first boot, Xen-Orchestra won't be
able to help you there to create your bond.

> - In case of an upgrade, your host previous configuration will be kept once the

Reviewer: ```suggestion
- For upgrades, your existing configuration will remain intact after the upgrade is done.
```

upgrade is done.

:::

Our installer does not support LACP, this guide explains how to workaround this
limitation, with some caveat, be sure to read all the explanations before
proceeding.

> When using LACP, your switch needs to be configured for it, and if you don't

Reviewer: ```suggestion
When using LACP, your switch needs to be configured for it, and if you cannot or prefer not to reconfigure your switch, you won’t be able to use the single-interface option provided by the installer.
```

want to (or cannot) reconfigure your switch, that can prevent you from using a
single interface that are offered by the installer.

For other bonding methods (active-backup, balance-lsb), the switch does not
need to be aware of the bond, and you can simply use one of the interfaces that
will be part of the bond.

## Do you actually need network?

> For these cases, you will need to have a working network configuration:

Reviewer: ```suggestion
A working network configuration is only necessary in the following scenarios:
```

- New installation from net install ISO
- Upgrade of supporting hosts *even from a full ISO*
- Upgrade of all hosts from net install ISO

> Network is not needed for:

Reviewer: ```suggestion
Network access is not required for:
```

- New installation from full ISO

> - Upgrade of master from full ISO

Reviewer: ```suggestion
- Upgrading the master host from the full ISO
```

## Installation or Upgrade steps

:::info
In the case of an upgrade, you could expect the installer to detect your
previous bond configuration, that is not the case as in XCP-ng the network
configuration is handled by XAPI and is applied to Open vSwitch, but the
installation process is not running either of those.
:::

> First, boot the on your installation media, you will reach the "Select Keymap"

Author: Done, I tried placing the screenshots properly as well and did rework sentences a bit to fit the way it is now presented, feel free to let me know if that works.

Simply select your keymap, and proceed to the next screen, "Welcome to XCP-ng
Setup", now the interfaces should have been renamed. You can change to the
second terminal using either `alt+F2` on `alt+right`, and confirm the
interfaces are named `ethX` using `ip link`. If that's the case, you can
proceed with the next steps.

Now we can configure a bond using standard linux tools.

```bash
modprobe bonding mode=802.3ad miimon=100 # automatically creates bond0 in 8.3 installer
ip link set eth0 down
ip link set eth1 down
ip link set eth0 master bond0
ip link set eth1 master bond0
```

For dhcp, use:

```bash
dhclient bond0
```

Or configure a static ip the usual way:

```bash
ip addr add aaa.bbb.ccc.ddd/NN dev bond0
ip route add default via aaa.bbb.ccc.1
ip link set bond0 up
```

![configure LACP](../assets/img/installer-creating-lacp-bond.png)

### Upgrading a supporting host

You can follow the standard install procedure as usual, when prompted for the
networking configuration, you should see an option "Use existing
configuration", select that option and finish your upgrade.

![use existing configuration](../assets/img/installer-use-existing-configuration.png)

> And you're all done, on reboot your host should gather back the bond

Reviewer: 'And' (or 'but') connects two clauses in a sentence. Avoid using them to start a brand new one.

```suggestion
You're all done! Your host will automatically restore its previous bond configuration after the reboot
```

configuration your host was using before the upgrade.

### New net install

During a net install you have 2 steps asking for network configuration:

> - a first one for the net install to be able to reach the installer repository

Reviewer: Feel free to use repetition: repetition makes it easier to remember an idea (that's what music is built on!)

```suggestion
- The first step configures the network so the installer can access the installation repository.
- The second step sets up your host’s network configuration after the installation is complete.
```

This wording also injects verbs into the bullet list, which makes sentences more concrete ("what do we actually do here?")

- a second one for your host configuration after the installation is complete

During the first step, you can tell the installer to "Use existing
configuration", and it should be able to reach the repository and move on with
the next steps of the installer.

![net install use existing configuration](../assets/img/netinstall-repo-network.png)

> On the second one, that option won't be shown, and you will have to use one of

Reviewer: While the reader is busy intalling XCP-ng, they may not always remember the previous steps. Again, repetition can solve this issue:

```suggestion
During the second step, the **Use existing configuration** option won’t be available. Instead, you’ll need to select one of the interfaces provided by the installer. While you can still configure the hostname and other basic network settings, you won’t be able to set up your bond at this stage.
```

the interfaces offered by the installer, that still allows you to configure the
hostname and some other network settings, but you won't be able to configure
your bond at this point.

![net install management network](../assets/img/netinstall-management-network.png)

> Proceed with the installation, and once it's complete, you can reboot your

Reviewer: - Just like Warnings, feel free to use tips. They can make useful information (that aren't actual instructions) stand out so the reader can spot it more easily.
- Again, numbered lists could be useful here. I'm not sure which step we would be at, right there, but it could look like this (see the code block below)
- Make sure the most important idea goes first, or at least make your ideas follow a logical order (e.g. from the cause to the consequence). Here, it's because the first boot can take a while that we advise users to give the host enough time. As a result, "The first boot can take a while" should go first. This way, if the first boot does not take a while, the reader isn't concerned by this part of the guide, and they can choose to immediately move on to the next step.

```
4. Proceed with the installation.
5. You can reboot your host.
:::tip
The first boot can take a while. Give the host enough time for everything to start up before proceedingto the next steps. the first boot can take a while.
:::
```

Author: I used numbered list, so given there are 2 paths (upgrade / net install), I restarted both with the step number following the common part, let me know if that seems right.

host, give it enough time for everything to startup before going to the next
steps, the first boot can take a while.

If you used dhcp, `xsconsole` will likely show you `No network configured`, if
you used a static IP it will show it, but on the `ethX` interface you chose.
You now need to create your bond and make it the management interface.

:::info

> `xe` commands now have a better completion, take advantage of it to avoid using

Reviewer: ```suggestion
`xe` commands now feature improved tab completion. Use it to avoid running `*-list` commands to find UUIDs and retyping them manually.
```

the `*-list` commands to find uuids and retyping them.
:::

First, create a new network named bond0 for example:

```bash
xe network-create name-label="bond0"
```

Create the bond using the network-uuid that command just returned:

```bash
xe bond-create mode=lacp network-uuid=<network_uuid> pif-uuids=<pif_uuid#1>,<pif-uuid#2>
```

Now, set it as your management interface:

```bash
xe pif-reconfigure-ip uuid=<bond_pif_uuid> netmask=255.255.255.0 gateway=192.168.1.1 IP=192.168.1.5 mode=static
xe host-management-reconfigure pif-uuid=<bond_pif_uuid>
```

> You should now be all set!

Reviewer: I think we can be a little more assertive here. If our guide is well made, the user should _definitely_ be all set:

```suggestion
You're all set!
```
